src/communication/network_interface_sr.c

```diff
@@ -7268,6 +7269,368 @@ slogtb_dump_trantable (THREAD_ENTRY * thread_p, unsigned int rid, char *request,
+	  buffer = (char *) db_private_alloc (thread_p, size);
+
+	  if (buffer == NULL)
+	    {
+	      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, size);
+	      error = ER_OUT_OF_VIRTUAL_MEMORY;
+	    }
+	  else
+	    {
+	      /* 3) packing information
+	       *    - packing information with sequence of
+	       *      module info -> component info w/ subcomponent info */
+	      ptr = buffer;
+
+	      for (int idx = 0; idx < module_count; idx++)
+		{
+		  ptr = or_pack_string (ptr, module_info[idx].name);
+		  ptr = or_pack_int64 (ptr, module_info[idx].stat.init_stat);
+		  ptr = or_pack_int64 (ptr, module_info[idx].stat.cur_stat);
+		  ptr = or_pack_int64 (ptr, module_info[idx].stat.peak_stat);
+		  ptr = or_pack_int (ptr, module_info[idx].stat.expand_count);
+
+		  ptr = or_pack_int (ptr, module_info[idx].num_comp);
+
+                  // *INDENT-OFF*
+                  for (const auto &comp_info : module_info[idx].comp_info)
+                    {
+                      ptr = or_pack_string (ptr, comp_info.name);
+                      ptr = or_pack_int64 (ptr, comp_info.stat.init_stat);
+                      ptr = or_pack_int64 (ptr, comp_info.stat.cur_stat);
+                      ptr = or_pack_int64 (ptr, comp_info.stat.peak_stat);
+                      ptr = or_pack_int (ptr, comp_info.stat.expand_count);
+
+                      ptr = or_pack_int (ptr, comp_info.num_subcomp);
+
+                      for (const auto &subcomp_info : comp_info.subcomp_info)
+                        {
+                          ptr = or_pack_string (ptr, subcomp_info.name);
+                          ptr = or_pack_int64 (ptr, subcomp_info.cur_stat);
+                        }
+                    }
+                  // *INDENT-ON*
+		}
+	    }
+	}
+
+      /* 4) deallocate memory */
+      db_private_free (thread_p, module_info);
+    }
+
+  if (error != NO_ERROR)
+    {
+      ptr = or_pack_int (reply, 0);
+      ptr = or_pack_int (ptr, error);
+      css_send_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply));
+    }
+  else
+    {
+      ptr = or_pack_int (reply, size);
+      ptr = or_pack_int (ptr, error);
+      css_send_reply_and_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply), buffer, size);
+    }
+  db_private_free_and_init (thread_p, buffer);
+}
+
+/*
+ * smmon_get_module_info_summary -
+ *
+ * return:
+ *
+ *   rid(in):
+ *   request(in):
+ *   reqlen(in):
+ */
+void
+smmon_get_module_info_summary (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int reqlen)
+{
+  char *buffer, *ptr;
+  int size = 0;
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  int error = NO_ERROR;
+  MMON_MODULE_INFO *module_info;
+  int module_count;
+
+  ptr = or_unpack_int (request, &module_count);
+
+  module_info = (MMON_MODULE_INFO *) malloc (sizeof (MMON_MODULE_INFO) * MMON_MODULE_LAST);
```

Comment:
Consider db_private_alloc() too.

Reply:
Nit: (Because of code consistency)

I think that smmon_get_*() function can use vector container iso malloc/free() or db_private_alloc/db_private_free(). Please consider it.